src/app/application-page/application-page.component.ts

```diff
@@ -23,4 +28,16 @@ export class ApplicationPageComponent implements OnInit {
     );
   }
 
+  changeStatus(status) {
+    this.applicationWithNewStatus  = RegisterFormComponent.apply(this.application$);
```

Comment:
That's an _interesting_ usage - what is the goal of this?
Try not to mix the presentational component (RegisterFormComponent in this case) and the actual data model, which I assume you are using here. If you want to have a constructor - then use a plain class to construct the  application - otherwise - just use an object.